### PR TITLE
APPS-306: Payment API improvements

### PR DESCRIPTION
**Pull request by olehnovatskyi-spryker**
- Developer(s): @olehnovatskyi-spryker

- Release Group: https://release.spryker.com/release-groups/view/3465

- Ticket: https://spryker.atlassian.net/browse/APPS-306

##### Change log

Improvements

- Introduced `PayoneFacade::orderPostSave()` to check log redirect and authorization redirect in one function call.
- Introduced `PayoneFacade::saveOrderPayment()`.
- Introduced `PayoneCheckoutDoSaveOrderPlugin` to save **Payone** payment.
- Introduced `PayoneCheckoutPostSavePlugin` to execute additional logic after saving order data to DB.
- Introduced `PayoneCheckoutPreConditionPlugin` to validate the quote before saving the order to DB.

Deprecations
- Deprecated `PayoneFacadeInterface::saveOrder()`.
- Deprecated `PayoneFacadeInterface::postSaveHook()`.
- Deprecated `PayoneFacadeInterface::executeCheckoutPostSaveHook()`.
- Deprecated `PayoneCheckoutPostSaveHookPlugin`.
- Deprecated `PayonePostSaveHookPlugin`.
- Deprecated `PayonePreCheckPlugin`.
- Deprecated `PayoneSaveOrderPlugin`.

### CODE REVIEW

**Review by oleksander-kiiashko — APPROVED** [review]

File: tests/SprykerEcoTest/Zed/Payone/_support/_generated/PayoneZedTesterActions.php [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<?php  //[STAMP] f481abe9db719f435b773d60b6f5c72a
+<?php  //[STAMP] 44c949f5ab6d2d78611c50300791b34e
```
Comment (matweew) [inline tests/SprykerEcoTest/Zed/Payone/_support/_generated/PayoneZedTesterActions.php:1]: please remove, it's generated one

File: src/SprykerEco/Zed/Payone/Communication/Plugin/Checkout/PayoneCheckoutPostSavePlugin.php [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+/**
+ * MIT License
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerEco\Zed\Payone\Communication\Plugin\Checkout;
+
+use Generated\Shared\Transfer\CheckoutResponseTransfer;
+use Generated\Shared\Transfer\QuoteTransfer;
+use Spryker\Zed\CheckoutExtension\Dependency\Plugin\CheckoutPostSaveInterface;
```
Comment (matweew) [inline src/SprykerEco/Zed/Payone/Communication/Plugin/Checkout/PayoneCheckoutPostSavePlugin.php:12]: this interface requires new version of "spryker/checkout-extension" which is absent in composer.json

File: tests/SprykerEcoTest/Zed/Payone/Business/Order/SaveOrderTest.php [hunk 3]
```diff
@@ -53,6 +53,32 @@ public function testSaveOrder()
         $this->assertEquals('12345', $paymentDetailEntity->getAmount());
     }
 
+    /**
+     * @return void
+     */
+    public function testSaveOrderPayment()
```
Comment (matweew) [inline tests/SprykerEcoTest/Zed/Payone/Business/Order/SaveOrderTest.php:59]:
```suggestion
    public function testSaveOrderPaymentStoresCorrectOrderPayment()
```

**Review by matweew — CHANGES_REQUESTED** [review]

File: src/SprykerEco/Zed/Payone/Business/Payment/PaymentManager.php [hunk 4]
```diff
@@ -1041,6 +1041,10 @@ public function isPaymentDataRequired(OrderTransfer $orderTransfer)
      */
     public function postSaveHook(QuoteTransfer $quoteTransfer, CheckoutResponseTransfer $checkoutResponse)
     {
+        if ($quoteTransfer->getPayment()->getPaymentProvider() !== PayoneConfig::PROVIDER_NAME) {
+            return $checkoutResponse;
+        }
```
[the remark this answers is not in the document]
Comment (olehnovatskyi-spryker) [reply src/SprykerEco/Zed/Payone/Business/Payment/PaymentManager.php:1046]: discussed

**Comment by matweew** [comment]
also please try to switch to GitHub Actions (example https://github.com/spryker-eco/after-pay/pull/23/files#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03f)

and remove .travis.yml

the goal to execute `phpcs` and if possible `phpstan`

File: tests/SprykerEcoTest/Yves/Payone/_support/_generated/PayoneYvesTesterActions.php [hunk 5]
```diff
@@ -1,4 +1,4 @@
-<?php  //[STAMP] 1a2fb5b7a87bd7041206e242f9058425
+<?php  //[STAMP] 78f011b48081081169267cd48d746928
```
[the remark this answers is not in the document]
Comment (matweew) [reply tests/SprykerEcoTest/Yves/Payone/_support/_generated/PayoneYvesTesterActions.php:1]: and add `tests/**/_generated/` to .gitignore

**Review by matweew — APPROVED** [review]

**Pull request merged**